std/os.zig

```diff
@@ -104,32 +104,63 @@ pub fn getrandom(buf: []u8) GetRandomError!void {
         return windows.RtlGenRandom(buf);
     }
     if (linux.is_the_target) {
-        while (true) {
-            const err = if (std.c.versionCheck(builtin.Version{ .major = 2, .minor = 25, .patch = 0 }).ok) blk: {
-                break :blk errno(std.c.getrandom(buf.ptr, buf.len, 0));
+        var buff = buf[0..];
```

Comment:
I think this loop and `buff` logic can be shared across all the targets?

Reply:
It can be shared across FreeBSD and Linux when using `getrandom()` on Linux. When using `/dev/urandom`, it's already handled by that function. When using WASI I believe (can't find any docs) the function always fills the buffer. When using Windows (`RtlGenRandom`) I believe it also always fills the buffer.

I can refactor to make it common to FreeBSD and Linux, will do that now.

Reply:
> When using Windows (`RtlGenRandom`) I believe it also always fills the buffer.

From https://github.com/ziglang/zig/issues/3015#issue-477007953 I got the idea that you would need to repeat perform multiple calls if the user asked for >4GB of data? i.e. the windows branch returns `math.min(math.maxInt(ULONG), requested_length)`?

Reply:
That issue is about the actual definition being wrong - we pass a possibly 64 bit sized `usize` to a Windows function that only takes a 32 bit sized `ULONG`. It will also have to loop, but for slightly different reasons and @andrewrk expressed he'd like the loop to be in the lower layer (`std/os/windows.zig` instead of `std/os.zig`) in that case.

Reply:
Anyway, I've updated the code to share the loop logic between FreeBSD and Linux, but haven't tested it myself yet - hopefully it passes CI.